doc/reference/reference_lua/msgpack.rst

```diff
@@ -526,3 +564,210 @@ and :ref:`YAML <yaml-cfg>`.
+    A C-style string pointer may be described as ``cdata<char *>`` or ``cdata<const char *>``.
+
+    :param buffer C_style_string_pointer: a pointer to a raw MsgPack string.
+    :param integer size: number of bytes in the raw MsgPack string.
+
+    :return: a MsgPack object
+
+    :rtype: userdata
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local buffer = require('buffer')
+        local buf = buffer.ibuf()
+        msgpack.encode({1, 2, 3}, buf)
+        local mp = msgpack.object_from_raw(buf.buf, buf:size())
+
+..  _msgpack-is-object:
+
+..  function:: is_object(some_argument)
+
+    Since version 2.10.1.
+
+    Check if the given argument is a MsgPack object.
+
+    :param some_agrument: any argument.
+
+    :return: ``true`` if the argument is a MsgPack object; otherwise, ``false``
+
+    :rtype: boolean
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local mp = msgpack.object(123)
+        msgpack.is_object(mp) -- returns true
+        msgpack.is_object({}) -- returns false
+
+..  _msgpack-object-methods:
+
+..  class:: msgpack_object
+
+    The MsgPack object has the following methods:
+
+    ..  method:: decode()
+
+        Since version 2.10.1.
+
+        Decode MsgPack data in a MsgPack object and return a Lua object.
+
+        :return: a Lua object
+
+        :rtype: Lua object
+
+    ..  method:: iterator()
+
+        Since version 2.10.1.
+
+        Return an iterator over the MsgPack data.
+
+        A MsgPack iterator object has its own :ref:`set of methods <msgpack-object-iterator-methods>`.
+
+        :return: an iterator object over the MsgPack data
+
+        :rtype: userdata
+
+    A MsgPack object can be passed to the MsgPack encoder with the same effect as passing the original Lua object:
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local mp = msgpack.object(123)
+        msgpack.object({mp, mp}):decode()         -- returns {123, 123}
+        msgpack.decode(msgpack.encode({mp, mp}))  -- returns {123, 123}
+
+    In particular, this means that if a MsgPack object stores an array, it can be inserted into a database space:
+
+    ..  code-block:: lua
+
+        box.space.my_space:insert(msgpack.object({1, 2, 3}))
+
+..  _msgpack-object-iterator-methods:
+
+..  class:: iterator_object
+
+    The MsgPack iterator object has the following methods:
+
+    ..  method:: decode_array_header()
+
+        Since version 2.10.1.
+
+        Decode a MsgPack array header under the cursor and return the number of elements in the array.
+        After calling this function, the iterator points to the first element of the array
+        or to the value following the array if the array is empty.
+
+        Raise an error if the type of the value under the iterator cursor is not ``MP_ARRAY``.
```

Comment:
AFAICS in other places we put errors under a separate sub-section, e.g. see "Possible errors" in https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/insert/

Reply:
Corrected.